src/DssSpell.sol

```diff
@@ -52,82 +59,161 @@ contract DssSpellAction is DssAction {
         return true;
     }
 
-    address internal immutable D3M_HUB                  = DssExecLib.getChangelogAddress("DIRECT_HUB");
-    address internal immutable D3M_MOM                  = DssExecLib.getChangelogAddress("DIRECT_MOM");
+    uint256 internal constant ZERO_PT_SEVEN_FIVE_PCT_RATE    = 1000000000236936036262880196;
+    uint256 internal constant ONE_PCT_RATE                   = 1000000000315522921573372069;
+    uint256 internal constant ONE_PT_SEVEN_FIVE_PCT_RATE     = 1000000000550121712943459312;
+    uint256 internal constant THREE_PT_TWO_FIVE_PCT_RATE     = 1000000001014175731521720677;
 
-    address internal constant SPARK_D3M_PLAN            = 0x104FaDbb7e17db1A685bBa61007DfB015206a4D2;
-    address internal constant SPARK_D3M_POOL            = 0xAfA2DD8a0594B2B24B59de405Da9338C4Ce23437;
-    address internal constant SPARK_D3M_ORACLE          = 0xCBD53B683722F82Dc82EBa7916065532980d4833;
+    uint256 internal constant MILLION                        = 10 ** 6;
+    uint256 internal constant WAD                            = 10 ** 18;
+    uint256 internal constant RAD                            = 10 ** 45;
 
-    address internal constant SPARK_ADAI                = 0x4DEDf26112B3Ec8eC46e7E31EA5e123490B05B8B;
-    address internal constant SPARK_DAI_STABLE_DEBT     = 0xfe2B7a7F4cC0Fb76f7Fc1C6518D586F1e4559176;
-    address internal constant SPARK_DAI_VARIABLE_DEBT   = 0xf705d2B7e92B3F38e6ae7afaDAA2fEE110fE5914;
-    address internal constant SPARK_POOL_CONFIGURATOR   = 0x542DBa469bdE58FAeE189ffB60C6b49CE60E0738;
+    // 01 May 2023 12:00:00 AM UTC
+    uint256 public constant MAY_01_2023                      = 1682899200;
+    // 01 May 2024 11:59:59 PM UTC
+    uint256 public constant MAY_01_2024                      = 1714607999;
+    // 01 May 2025 11:59:59 PM UTC
+    uint256 public constant MAY_01_2025                      = 1746143999;
```

Comment:
small comment here that if you use 11:59:59 you should probably take the day beforehand (see previous spells that do this).

Reply:
Thanks, fixed.